### ISSUE TITLE
Support chained linked servers

### ISSUE THREAD

**Issue by n3rada** [skahwah/SQLRecon#16]
Current functionality in SQLRecon does not support the execution of queries or commands across multiple linked servers in a chain. This limitation restricts the ability to perform reconnaissance or execute actions on a target server that is only accessible through a chain of linked servers.

## Scenario
Consider a scenario where there are multiple SQL servers linked sequentially, such as:
```txt
SQL01 -> SQL02 -> SQL03
```
In this configuration, the goal is to execute commands or perform reconnaissance on SQL03, starting from SQL01, utilizing the user credentials on SQL01. The user needs to "hop" through SQL02 to reach SQL03.

## Current Limitation
As of the current version, SQLRecon does not support the chaining of linked servers to execute commands or queries on a server that is not directly linked to the initial server (SQL01 in this scenario).

## Feature Enhancement
I propose enhancing the command parser in SQLRecon to support an argument list that defines a chain of linked servers. This enhancement would allow users to specify a sequence of servers through which SQLRecon should navigate to reach the final target server.

The enhancement could introduce a new way to specify linked servers in sequence, allowing SQLRecon to chain these links together:
```shell
sqlrecon /a:local /h:SQL01 /u:test01 /p:aaaaccccc /l:SQL02 /l:SQL03 /m:llinks
```
Where `/l:SQL02` and `/l:SQL03` define the chain of linked servers to traverse.

## Outcome
With such an enhancement, the tool should be able to execute the specified action on SQL03 by leveraging the linked server connections from SQL01 to SQL02 and then from SQL02 to SQL03. This capability will significantly extend SQLRecon's usefulness in environments with complex linked server architectures.

**Comment by skahwah** [comment]
Thanks for your suggestion. In an earlier version of SQLRecon I somewhat implemented a solution to solve the double-hop problem. There are several issues to consider when you're executing queries to a secondary linked server:

1. Privilege context - what authentication context should you carry through the entirety of the SQL transaction.
2. RPC - certain operations (`EXEC AT` in particular) will require RPC to be enabled on the final destination server, enabling this is possible, but requires the correct privilege context.
3. OpenQuery Limitations - To my knowledge, it's not possible to achieve a double hop execution cleanly with OpenQuery. It is [sort of possible to do this with EXEC](https://dba.stackexchange.com/questions/78196/how-to-query-from-a-linked-server-via-a-linked-server-in-sql-server), but again this requires the right privilege context.

It might be possible to implement a double hop query for basic SQL queries that don't involve any sort of command execution or escalation style attacks.

Hopefully this all makes sense. I'm interested to know if you have a solution for this issue!

**Comment by n3rada** [comment]
I succeeded in executing multiple hop queries though `openquery`, I let you take a look at my last Pull Request ! 😊

I'm keen on adding a better `AT` method like you have done with `RPC`. I am waiting for you to approve my PR before diving into this among others.

**Comment by skahwah** [comment]
@n3rada I appreciate your contributions and detailed documentation. Give me some time to review and test your changes before I merge in your pull request. Thank you!

**Comment by skahwah** [comment]
Included in v3.7. Thanks!
